packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -154,18 +155,33 @@ function wrap(middleware, handleNoSuchObject = false) {
 async function _getDashboardStats(app) {
   const dashboard = {}
 
+  let hvSupportedVersions
+  try {
+    hvSupportedVersions = await app.apiMethods['xoa.getHVSupportedVersions']()
+  } catch (e) {
+    hvSupportedVersions = {
+      'Citrix Hypervisor': '>=8.2.0',
+      'XCP-ng': '>=8.2.0',
+    }
+  }
+
   const poolIds = new Set()
   const hosts = []
+  let nHostsEol = 0
 
   for (const obj of app.objects.values()) {
     if (obj.type === 'host') {
       hosts.push(obj)
       poolIds.add(obj.$pool)
+      if (!semver.satisfies(obj.version, hvSupportedVersions[obj.productBrand])) {
```

Comment:
What if `hvSupportedVersions[obj.productBrand]` is `undefined`?

Reply:
Then `semver.satisfies` will return `false` and we will go into the `if` condition to count the host as `EOL`.
BTW, `hvSupportedVersions[obj.productBrand]` is not supposed to be `undefined` since we are covering `CH` and `XCP-ng`

Reply:
Still, this case should be handled to avoid issues in the future.

If `semver.satisfies` returns `false` for `undefined`, it's good enough for me.

Reply:
it's the case